include/deal.II/numerics/matrix_creator.templates.h

```diff
@@ -1332,94 +1335,97 @@ namespace MatrixCreator
             if (fe_is_system)
               // FE has several components
               {
+                coefficient_vector_values.resize (fe_values.n_quadrature_points,
+                                                  Vector<number>(n_components));
+                coefficient_values.resize (fe_values.n_quadrature_points, 1.);
+
                 rhs_values_system.resize (fe_values.n_quadrature_points,
                                           Vector<number>(n_function_components));
                 boundary_functions.find(cell->face(face)->boundary_id())
                 ->second->vector_value_list (fe_values.get_quadrature_points(),
                                              rhs_values_system);
+                if (coefficient_is_vector)
+                  // In case coefficient is vector-valued, fill
+                  // all components
+                  coefficient->vector_value_list (fe_values.get_quadrature_points(),
+                                                  coefficient_vector_values);
+                else
+                  // In case the scalar function is given, update the
+                  // values, if not - use the default (1.0)
+                  {
+                    if (coefficient != nullptr)
+                      coefficient->value_list (fe_values.get_quadrature_points(),
+                                               coefficient_values);
 
-                if (coefficient != nullptr)
+                    for (unsigned int point=0; point<fe_values.n_quadrature_points; ++point)
+                      coefficient_vector_values[point] = coefficient_values[point];
+                  }
+
+                // Special treatment for Hdiv and Hcurl elements,
+                // where only normal or tangential components
+                // should be projected
```

Comment:
Can you extend this by saying how exactly you will be doing this? I have to admit that I don't quite understand the code that follows, so adding a statement that explains the exact adjustment would be useful.

Also, would you mind extending the documentation of the function in the .h file by a note that states how exactly the mass matrix is defined?